### PR TITLE
Micro ROS setup in APPTurtle

### PR DESCRIPTION
**Pull request by gabryelreyes**
Documentation and implementation of the Micro ROS components into the turtle application.

- Succesfully subscribes to a topic and prints the incoming data.
- Only available for the Target. Simulation cannot be build with the current version.
- Uses a fork of the micro_ros_platformio library to fix issues in the CI tests. See [here](https://github.com/gabryelreyes/micro_ros_platformio/commit/27ea8cae8977061287014a69e6864ed2a356a9e5) for the changes.

### CODE REVIEW

File: lib/APPTurtle/src/App.cpp [hunk 1]
```diff
@@ -55,6 +57,11 @@
  * Types and classes
  *****************************************************************************/
 
+/**
+ * Subscriber for Geometry Twist ROS messages.
+ */
+typedef Subscriber<geometry_msgs__msg__Twist> TwistSubscriber;
```
Comment (nhjschulz) [inline lib/APPTurtle/src/App.cpp:63]: Is this really the name to use ? Maybe use local typedef to turn int into something pretty.

File: lib/APPTurtle/src/App.cpp [hunk 2]
```diff
@@ -122,9 +132,52 @@ void App::setup()
...
+                {
+                    LOG_ERROR("TwistCallback received nullptr.");
+                }
+                else
+                {
+                    LOG_DEBUG("Twist data received.");
+
+                    /* Short blink to indicate reception. */
+                    Board::getInstance().getBlueLed().enable(true);
+                    delay(50U);
```
Comment (nhjschulz) [inline lib/APPTurtle/src/App.cpp:155]: Should we really delay ? 
We could turn it off when we finishedf processing the message.

File: lib/APPTurtle/src/Subscriber.h [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+static void localCallback(const void* msgData, void* subscriberContext);
```
Comment (nhjschulz) [inline lib/APPTurtle/src/Subscriber.h:59]: Doxyggen comment missing

File: lib/APPTurtle/src/Subscriber.h [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+     * Not allowed.
+     */
+    Subscriber();
+};
+
+/******************************************************************************
+ * Functions
+ *****************************************************************************/
+
+/**
```
Comment (nhjschulz) [inline lib/APPTurtle/src/Subscriber.h:238]: move comment to prototype at beginning of file

**Review by nhjschulz — CHANGES_REQUESTED** [review]

File: lib/APPTurtle/src/Subscriber.h (hunk 3 above)
Comment (gabryelreyes) [reply lib/APPTurtle/src/Subscriber.h:59]: It is documented at the definition and not in the declaration. Pathfinder rule :D

File: lib/APPTurtle/src/Subscriber.h (hunk 4 above)
Comment (gabryelreyes) [reply lib/APPTurtle/src/Subscriber.h:238]: See comment on the declaration

File: lib/APPTurtle/src/App.cpp (hunk 1 above)
Comment (gabryelreyes) [reply lib/APPTurtle/src/App.cpp:63]: Name is not pretty, as it is only used to create the subscription. Still made it local

File: doc/ROS2/setup/Target.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Run micro-ROS on ESP32 Target
+How setup and use micro-ROS DCS target `TurtleTarget`
+
+## Build TurtleTarget
+Perform standard pio build of the TurtleTarget application. It will pull the [micro_ros_platformio](https://github.com/micro-ROS/micro_ros_platformio) as library depenency and build scripts.
```
[the remark this answers is not in the document]
Comment (hoeftjch) [reply doc/ROS2/setup/Target.md]: changed link to forked repo.

[the remark this answers is not in the document]
Comment (hoeftjch) [reply doc/ROS2/setup/Target.md]: fixed

File: doc/ROS2/setup/Target.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+## bind USB port with WSL
```
[the remark this answers is not in the document]
Comment (hoeftjch) [reply doc/ROS2/setup/Target.md]: fixed

File: lib/APPTurtle/src/App.cpp (hunk 2 above)
Comment (gabryelreyes) [reply lib/APPTurtle/src/App.cpp]: Fixed

**Review by nhjschulz — APPROVED** [review]

**Pull request merged**